### PR TITLE
DOC: Confound enhancement documentation

### LINKED ISSUE #1616: DOC: Improve descriptions of new confounds model
Include revisions suggested in #1586 regarding documentation of that PR.

### PR DESCRIPTION
**Pull request by rciric**
## Changes proposed in this pull request

* Add documentation for confound enhancement
* Resolves #1616 

## Documentation that should be reviewed

* `outputs.rst` is extended to include more complete descriptions of the new confounds functionality.

### CODE REVIEW

File: docs/outputs.rst [hunk 1]
```diff
@@ -177,17 +177,59 @@ additional noise components are calculated using :abbr:`CompCor (Component Based
...
+      "Mask": "combined",
+      "Method": "aCompCor",
+      "Retained": false,
+      "SingularValue": 20.7955177198,
+      "VarianceExplained": 0.0701465624
+    }
+  }
+
+For CompCor decompositions, entries include:
+- ``Method``: anatomical or temporal CompCor
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion

  - ``Method``: anatomical or temporal CompCor.
```

File: docs/outputs.rst [hunk 2]
```diff
@@ -177,17 +177,59 @@ additional noise components are calculated using :abbr:`CompCor (Component Based
...
+- ``Mask``: denotes the ROI where the decomposition that generated the component was performed: ``CSF``, ``WM``, or ``combined`` for anatomical CompCor
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
  - ``Mask``: denotes the ROI where the decomposition that generated the component
    was performed: ``CSF``, ``WM``, or ``combined`` for anatomical CompCor.

File: docs/outputs.rst [hunk 3]
```diff
@@ -177,17 +177,59 @@ additional noise components are calculated using :abbr:`CompCor (Component Based
...
+- ``SingularValue``: Singular value of the component
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
  - ``SingularValue``: Singular value of the component.
```

File: docs/outputs.rst [hunk 4]
```diff
@@ -177,17 +177,59 @@ additional noise components are calculated using :abbr:`CompCor (Component Based
...
+- ``VarianceExplained``: fraction of variance explained by the component across the decomposition ROI mask
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
  - ``VarianceExplained``: the fraction of variance explained by the component across the decomposition ROI mask.
```

File: docs/outputs.rst [hunk 5]
```diff
@@ -177,17 +177,59 @@ additional noise components are calculated using :abbr:`CompCor (Component Based
...
+- ``CumulativeVarianceExplained``: total fraction of variance explained by the component and all preceding components
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
  - ``CumulativeVarianceExplained``: the total fraction of variance explained by this particular component
    and all preceding components.
```

File: docs/outputs.rst [hunk 6]
```diff
@@ -177,17 +177,59 @@ additional noise components are calculated using :abbr:`CompCor (Component Based
...
+- ``Retained``: Indicates whether the component was saved in ``desc-confounds_regressors.tsv`` for use in denoising. Entries that are not saved in the data file for denoising are still stored in metadata with the ``dropped`` prefix.
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
  - ``Retained``: Indicates whether the component was saved in ``desc-confounds_regressors.tsv``
    for use in denoising.
    Entries that are not saved in the data file for denoising are still stored in metadata with the
    ``dropped`` prefix.

```

File: fmriprep/workflows/bold/confounds.py [hunk 7]
```diff
@@ -161,13 +161,18 @@ def init_bold_confs_wf(
 in T1w space, after their projection to the native space of each
 functional run (using the inverse BOLD-to-T1w transformation). Components
 are also calculated separately within the WM and CSF masks.
+For each CompCor decomposition, the k components with the largest singular
```
Comment (oesteban) [inline fmriprep/workflows/bold/confounds.py]:
```suggestion
For each CompCor decomposition, the :math:`k` components with the largest singular
```

File: docs/outputs.rst [hunk 8]
```diff
@@ -177,17 +177,59 @@ additional noise components are calculated using :abbr:`CompCor (Component Based
...
+denoising. The original Behzadi aCompCor implementation ([Behzadi2007]_) can be applied using
+components from the combined ROI, while the more recent Muschelli implementation
+([Muschelli2014]_) can be applied using the WM and CSF ROIs. To determine the provenance of
+each component, consult the metadata file (see below).
 
 All these confounds can be used to perform *scrubbing* and *censoring* of outliers,
 in the subsequent first-level analysis when building the design matrix,
-and in group level analysis.
+and in group level analysis. *Spike regressors* for outlier censoring can also be
+generated from within fMRIPrep using the command line options `--fd-spike-threshold`
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
generated from within fMRIPrep using the command line options ``--fd-spike-threshold``
```

File: docs/outputs.rst [hunk 9]
```diff
@@ -177,17 +177,59 @@ additional noise components are calculated using :abbr:`CompCor (Component Based
...
+and `--dvars-spike-threshold`. Spike regressors are stored in separate `motion_outlier_XX`
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
and ``--dvars-spike-threshold``. Spike regressors are stored in separate ``motion_outlier_XX``
```

File: docs/outputs.rst [hunk 10]
```diff
@@ -203,8 +245,50 @@ An example of these plots follows:
     in cortical gray matter regions (blue), subcortical gray matter (orange),
     cerebellum (green) and the union of white-matter and CSF compartments (red).
 
+Noise components computed during each CompCor decomposition are evaluated according
+to the fraction of variance that they explain across the nuisance ROI.
+This is used by fMRIPrep to determine whether each component should be saved for
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
This is used by *fMRIPrep* to determine whether each component should be saved for
```

File: docs/outputs.rst [hunk 11]
```diff
@@ -203,8 +245,50 @@ An example of these plots follows:
...
+use in denoising operations: a component is saved if it contributes to explaining
+the top 50 percent of variance in the nuisance ROI. fMRIPrep can be configured to
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
the top 50 percent of variance in the nuisance ROI.
*fMRIPrep* can be configured to
```

File: docs/outputs.rst [hunk 12]
```diff
@@ -203,8 +245,50 @@ An example of these plots follows:
...
+save all components instead using the command line option ``--return-all-components``.
+FMRIPrep reports include a plot of the cumulative variance explained by each
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
*fMRIPrep* reports include a plot of the cumulative variance explained by each
```

File: docs/outputs.rst [hunk 13]
```diff
@@ -203,8 +245,50 @@ An example of these plots follows:
...
+component, ordered by descending singular value.
+
+.. figure:: _static/sub-01_task-rest_compcor.svg
+    :scale: 100%
+
+    The figure displays the cumulative variance explained by components for each
+    of four CompCor decompositions (left to right: anatomical CSF mask, anatomical
+    white matter mask, anatomical combined mask, temporal). The number of
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
    white matter mask, anatomical combined mask, temporal).
    The number of components is plotted on the abscissa and
```

File: docs/outputs.rst [hunk 14]
```diff
@@ -203,8 +245,50 @@ An example of these plots follows:
...
+    components is plotted on the abscissa and the cumulative variance explained
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
    the cumulative variance explained on the ordinate.
```

File: docs/outputs.rst [hunk 15]
```diff
@@ -203,8 +245,50 @@ An example of these plots follows:
...
+    on the ordinate. Dotted lines indicate the minimum number of components
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
    Dotted lines indicate the minimum number of components necessary
```

File: docs/outputs.rst [hunk 16]
```diff
@@ -203,8 +245,50 @@ An example of these plots follows:
...
+    necessary to explain 50%, 70%, and 90% of variance in the nuisance mask. By
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
    to explain 50%, 70%, and 90% of the variance in the nuisance mask.
```

File: docs/outputs.rst [hunk 17]
```diff
@@ -203,8 +245,50 @@ An example of these plots follows:
...
+    default, only the components that explain the top 50% of variance are saved.
```
Comment (oesteban) [inline docs/outputs.rst]:
```suggestion
    By default, only the components that explain the top 50% of the variance
    are saved.
```

Comment (oesteban) [reply docs/outputs.rst]: Is it possible to change this threshold of 50% with some flag or some other mechanism?

File: docs/outputs.rst [hunk 18]
```diff
@@ -203,8 +245,50 @@ An example of these plots follows:
...
+
+Also included is a plot of correlations among confound regressors. This can be
```
Comment (oesteban) [inline docs/outputs.rst]: Please start new sentences with newline (sorry I'll stop providing suggestions for this)

File: docs/outputs.rst (hunk 17 above)
Comment (rciric) [reply docs/outputs.rst]: Not possible right now, but it wouldn't take much effort to change the `--return-all-components` option to `--compcor-variance-threshold` with this behaviour.

File: fmriprep/workflows/bold/confounds.py [hunk 19]
```diff
@@ -161,13 +161,18 @@ def init_bold_confs_wf(
 in T1w space, after their projection to the native space of each
 functional run (using the inverse BOLD-to-T1w transformation). Components
 are also calculated separately within the WM and CSF masks.
+For each CompCor decomposition, the :math:`k` components with the largest singular
```
Comment (oesteban) [inline fmriprep/workflows/bold/confounds.py]: Sorry about this. I did not think it was on the boilerplate. Reverting...
```suggestion
For each CompCor decomposition, the k components with the largest singular
```

**Pull request merged**